checkov/dockerfile/image_referencer/provider.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+
+import os
+from typing import TYPE_CHECKING, Callable, Any
+
+from checkov.common.images.image_referencer import Image
+from checkov.common.util.str_utils import removeprefix
+from checkov.dockerfile.utils import DOCKERFILE_STARTLINE, DOCKERFILE_ENDLINE
+
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
+_ExtractImagesCallableAlias: TypeAlias = Callable[["dict[str, Any]"], "list[str]"]
+
+
+class DockerfileProvider:
+    __slots__ = ("definitions", "supported_resource_types")
```

Comment:
Is `supported_resource_types` used? Didn't find an initialization of it

Reply:
good old copy paste left over 😄